static/scripts/assignment_table.js

```diff
@@ -0,0 +1,481 @@
+            colHeaders: allHeaders,
+            columns: columns,
+            colWidths: 100,
+            columnHeaderHeight: 225,
+            rowHeaders: fixedRowsText,
+            rowHeaderWidth: 125,
+            // Disable the cells that contain user preferences
+            cells: function (row, col) {
+                const cellProperties = {};
+                if (row >= lenFixedRowsText && col >= lenFixedHeaders && row % 2 === 0) {
+                    cellProperties.readOnly = true;
+                }
+                return cellProperties;
+            },
+            afterInit: function () {
+                const sourceData = this.getSourceData();
+                saved_data.forEach(assignment => {
+                    const rowIndex = sourceData.findIndex(row => row.researchers.id === assignment.researcher_id);
+
+                    if (rowIndex !== -1) {
+                        const colIndex = this.propToCol(assignment.course_id);
+
+                        const commentsPlugin = this.getPlugin('comments');
+                        //Id is set for the two merged rows, and we want to set the comment for the assignment row
+                        commentsPlugin.setCommentAtCell(rowIndex + 1, colIndex, assignment.comment);
+                    }
+                });
+            },
+            afterGetColHeader: function (col, th) {
+                th.style.transform = 'rotate(180deg)';
+                th.style.writingMode = 'vertical-lr';
+                th.style.textAlign = 'center';
+                th.style.fontWeight = 'bold';
+                th.style.height = '100px';
+                th.style.lineHeight = '100px';
+
+                if (col > lenFixedHeaders - 1) {
+                    let colData = this.getDataAtCol(col);
+                    if (colData[RowIndices.TOTAL_ASSISTANT_NOW] >= colData[RowIndices.ASSISTANTS]) {
+                        th.style.backgroundColor = 'green';
+                    }
+                }
+            },
+            afterChange: function (changes) {
+                if (changes) {
+                    changes.forEach(([row, prop, oldValue, newValue]) => {
+                        let col = this.propToCol(prop);
+
+                        if ((col >= lenFixedHeaders && row >= lenFixedRowsText) && (row % 2 === 1)) {
+                            const colInfos = this.getDataAtProp(prop);
+                            const quadri = colInfos[RowIndices.QUADRI];
+                            let loadKey = quadri === 1 ? ColumnIndices.LOAD_Q1 : ColumnIndices.LOAD_Q2;
+
+                            //Count the number of assignments to determine the value of loads
+                            let researcherRow = this.getDataAtRow(row).slice(lenFixedHeaders);
+                            let loadValue = researcherRow.filter(value => value !== null && value !== '').length;
+                            this.setDataAtCell(row - 1, loadKey, loadValue);
```

Comment:
There is a big problem the way the total load for a researcher behaves. When assigning a course, you count all the non selected cells and apply the value to the quadri load of the course you just assigned. For example, when you assign a course from Q2 to a researcher, all his assignations are counted (Q1 and Q2) and the value is set to his Q2 total load.

You should have a way to only count the assignations from the correct semester (quadri).

[Screencast from 11-07-24 15:31:14.webm](https://github.com/user-attachments/assets/75f147a2-f7ca-4e03-a923-55c99a325bd0)

The same applies when removing assignations.

Reply:
Also, when indicating teacher names you only use the last name (here numbers). If multiple teachers have the last first name there is no way to differentiate them (yes, that happens). Either write the full names or add the first letter of their name. The second would not be perfect but would greatly reduce the risk.